content/manual/adoc/en/framework/gui_framework/own_components/own_component_samples/vaadin_addon_sample.adoc

```diff
@@ -7,63 +7,30 @@ This is an example of using the *Stepper* component available at `http://vaadin.
 
 Create a new project in CUBA Studio and name it `addon-demo`.
 
-A Vaadin add-on may be integrated if the application project has a *web-toolkit* module. Create the module by clicking the *Create web toolkit module* link of the *Project properties* navigator section.
+A Vaadin add-on may be integrated if the application project has a *web-toolkit* module. A convenient way to do this is to use *CUBA Studio*: in the main menu, click *CUBA > Advanced > Manage modules > Create 'web-toolkit' Module*.
 
-// TODO update Studio flow
-Then click the *New UI component* link. The *UI component generation* page will open. Select the *Vaadin add-on* value in the *Component type* section.
+Then add add-on dependencies:
 
-image::ui_component/studio_vaadin_addon_wizard_no_gui.png[align="center"]
-
-Fill in the following fields:
-
-* The *Add-on Maven dependency* field contains Maven coordinates of the Vaadin add-on. The add-on will be included as a dependency to the project. You can define coordinates in two formats:
-
-. As an XML copied from the add-on web site (http://vaadin.com/addon/stepper):
-+
-[source, xml]
-----
-<dependency>
-   <groupId>org.vaadin.addons</groupId>
-   <artifactId>stepper</artifactId>
-   <version>2.2.2</version>
-</dependency>
-----
-
-. In one line as you add dependencies in <<build.gradle,build.gradle>>: `org.vaadin.addons:stepper:2.2.2`
-
-* The *Inherited widgetset* field contains a widgetset name of the add-on:
-+
-[source]
-----
-org.vaadin.risto.stepper.widgetset.StepperWidgetset
-----
-
-* *Integrate into generic UI* - deselect this checkbox as we do not integrate the component into the Generic UI in this example.
-
-Press the *OK* button.
-
-If you open the project in the IDE, you can see that Studio has changed two files:
-
-. In `build.gradle`, the *web* module now contains a dependency on the add-on that contains the component.
+. In <<build.gradle,build.gradle>>, for the *web* module add a dependency on the add-on that contains the component:
 +
 [source, java]
 ----
 configure(webModule) {
     ...
     dependencies {
         ...
-        compile("org.vaadin.addons:stepper:2.2.2")
+        compile("org.vaadin.addons:stepper:2.4.0")
     }
 ----
 
-. The `AppWidgetSet.gwt.xml` file of the *web-toolkit* module now inherits the add-on widgetset:
+. In `AppWidgetSet.gwt.xml` file of the *web-toolkit* module, add an inheritance on the add-on widgetset:
```

Comment:
In `AppWidgetSet.gwt.xml` file of the *web-toolkit* module, indicate that the project's widgetset inherits from the add-on widgetset